### PR TITLE
feat: Add support for model-loading

### PR DESCRIPTION
**Pull request by MoritzWeber0**
IMPORTANT! This version in only compatible with Capella Docker images >= v2.0.0 (https://github.com/DSD-DBS/capella-dockerimages/pull/275)!
Make sure to migrate the Capella Docker images before.

Read-only sessions usually clone Git repositories before the session starts serving. This was implemented in the tool containers.

Since the behaviour is always the same, we'll move the behaviour to the Collaboration Manager.

Resolves https://github.com/DSD-DBS/capella-collab-manager/issues/1450

Changes: 
- The values.yaml and backend configuration have a different handling of registries now. You have to update the value "docker.registry" in your development backend configuration. If you don't have internal access, you tune those in the values.yaml: 
  - A new sessions registry was added (ghcr.io/dsd-dbs/capella-dockerimages), was the internal registry previously. It's used for the initialization of tools. After the first initialization, it can be changed in the UI.
  - The internal registry points to the Capella Collaboration Manager registry now (ghcr.io/dsd-dbs/capella-collab-manager)
- The frontend, backend, guacamole and docs image names are no longer customizable and are fetched from the internal registry instead. This change was introduced to reduce initial setup complexity.
- A new session-preparation image was added. The image is executed before each session start as init container and can be used for tool-independent session preparation actions.
- The Collaboration Manager clones all selected repositories for read-only sessions automatically and passed it as volume to the main container. Possible code in the tool containers that does the same can be removed.
- Read-only image support was removed. Instead, the persistent image is used and an empty workspace is mounted.

TODO: 

- [x] Drop read-only image in capella-dockerimages (https://github.com/DSD-DBS/capella-dockerimages/pull/275)
- [x] Handle duplicated project names in capella-dockerimages
- [x] Create missing .project file in capella-dockerimages
- [x] Database migration, remove read-only image and add environment variable

### CODE REVIEW

**Review by dominik003 — APPROVED** [review]
LGTM! Just a few minor comments that should be addressed before merging.

File: backend/capellacollab/sessions/hooks/interface.py [hunk 1]
```diff
@@ -33,6 +33,8 @@ class ConfigurationHookResult(t.TypedDict):
     environment: t.NotRequired[t.Mapping]
     volumes: t.NotRequired[list[operators_models.Volume]]
     warnings: t.NotRequired[list[core_models.Message]]
+    init_volumes: t.NotRequired[list[operators_models.Volume]]
+    init_environment: t.NotRequired[t.Mapping]
```
[the remark this answers is not in the document]
Comment (MoritzWeber0) [reply backend/capellacollab/sessions/hooks/interface.py:37]: Is fixed. We definitely have to introduce pydocstyle at some point.

File: backend/capellacollab/sessions/operators/models.py [hunk 2]
```diff
@@ -30,3 +31,11 @@ class PersistentVolume(Volume):
 
 class EmptyVolume(Volume):
     """An empty volume without content that is mounted into the container."""
+
+
+@dataclasses.dataclass
+class Container:
```
[the remark this answers is not in the document]
Comment (MoritzWeber0) [reply backend/capellacollab/sessions/operators/models.py:37]: You are right, doesn't seem to be used. Removed it.

File: backend/tests/sessions/hooks/test_session_preparation.py
[the remark this answers is not in the document]
Comment (MoritzWeber0) [reply backend/tests/sessions/hooks/test_session_preparation.py:1]: Done 👍

**Pull request merged**